### ISSUE TITLE
Unable to send simple verification email

### ISSUE THREAD

**Issue by edi** [kreait/firebase-php#412]
**Describe the bug**
I've upgraded to the latest version, which seems to have support for email verfication delivery. I get this error every time I'm trying to send the email (with no action settings)

```php
PHP Fatal error:  Uncaught Kreait\Firebase\Exception\RuntimeException: Custom Token Generation is disabled because the current credentials do not permit it in /firebase-php/vendor/kreait/firebase-php/src/Firebase/Auth/DisabledLegacyCustomTokenGenerator.php:28
```

**To Reproduce**
A self-contained code example to reproduce the behavior

```php
$factory = new \Kreait\Firebase\Factory;

$auth = $factory->createAuth();

$account = $auth->createUser([
  'email' => $data['email'],
  'password' => $data['password']
]);

$auth->sendEmailVerificationLink($data['email']);
```

**Expected behavior**
Trigger a verification email, just like the JS SDK does.

**Environment (please complete the following information):**
 - OS: Debian 8.11
 - PHP version: 7.3.16
 - Firebase SDK Version: 5.1.0

**Additional context**
I am not trying to send any custom OOB code or emails. Just a simple trigger. User creation works fine. Service account constant is defined.

**Comment by jeromegamez** [comment]
The error message is quite revealing :)

> Custom Token Generation is disabled because the current credentials do not permit it

You don‘t have the factory initialized with Service Account credentials. Please follow https://firebase-php.readthedocs.io/en/stable/setup.html

**Comment by jeromegamez** [comment]
Just for clarification - you seem to be authenticated since user creation worked. But it doesn‘t seem to be with service account credentials, which are currently needed by the SDK to send the OOB mails.

**Comment by edi** [comment]
Ah ok. That was the confusing part, because in the [documentation](https://firebase-php.readthedocs.io/en/stable/setup.html#with-autodiscovery), it's mentioned it can automatically discover the service account file from those constants, one of which is defined on my end `GOOGLE_APPLICATION_CREDENTIALS`. Basically contains the path to the service account file.

Will try initializing it with the file manually, see if it works :)

**Comment by edi** [comment]
Alright, so. One thing to mention here. While the above is technically my fault, I have all my configs set as environment vars, including the `GOOGLE_APPLICATION_CREDENTIALS` path.

Therefore, I had to use the following, because it couldn't automatically pick it up.
```php
$factory = (new Factory)->withServiceAccount( getenv('GOOGLE_APPLICATION_CREDENTIALS') );
```

**Comment by jeromegamez** [comment]
Please let me know if that worked, because you‘re right, if you had your credentials configured in environment variables, they should have been auto-discovered!

The issue could also be that the application default credentials don‘t specify a service account but user refresh credentials (usually after a `gcloud auth application-default login`). This was previously not possible at all and has been partially addressed after https://github.com/kreait/firebase-php/issues/399 came up.

User refresh credentials are now partially supported, but they aren‘t enough for many operations.

In addition, the sending of email verification links is currently broken (probably) on the API side - contrary to other OOB operations, those requests need to be authenticated with the recipient‘s ID token. To fetch the ID token, the SDK needs to create a custom token, and to create a custom token, it needs to sign it with the private key of a service account. 

This is all probably too much information if the issue is „just“ that the auto-discovery didn‘t work as expected, but I thought it wouldn‘t hurt as well 😅

**Comment by edi** [comment]
@jeromegamez Everything worked fine using the following:

```php
->withServiceAccount( getenv('GOOGLE_APPLICATION_CREDENTIALS') );
```

The auto discovery feature did not manage pulling it off previously, as it returned the error mentioned in the initial post.

Regarding the service account, it's a fully-fledged service account downloaded from firebase's project settings. It managed generating a custom token and sign it with the private key, then gracefully sent the email. 

Token worked just fine, which made me learn something new today. I could use your library to create my own custom verification email sender / handler :)

**Comment by jeromegamez** [comment]
Do it 💪😅

I just tested this as well and can confirm your experience. I'll try to find the issue and provide a fix soon!

**Comment by edi** [comment]
@jeromegamez Hope it helps, as it wasn't a biggie to deal, nor there is any backwards compatibilty to ware about, therefore all tests should remain untouched, unless I'm missing something?

Regards!

**Comment by jeromegamez** [comment]
Merged with 797ffb7292dd478bc533989afaeb438b6eae1780, released with [v5.1.1](https://github.com/kreait/firebase-php/releases/tag/5.1.1) 🚀

**Comment by carlitoscarrara** [comment]
Hello Jerome,

I am facing the same issue auto-discover issue running firebase-php v4.44.0

I have upgraded to 5.8.1 and 5.14.1 and the issue persists.

The issue only occurs on a DigitalOcean Lemp Droplet Server. No problem in Windows localhost.
